cmd/kubeadm/app/preflight/checks.go

```diff
@@ -869,6 +870,25 @@ func (ncc NumCPUCheck) Check() (warnings, errorList []error) {
 	return warnings, errorList
 }
 
+// MemCheck checks if the number of megabytes of memory is not less than required
+type MemCheck struct {
+	Mem uint64
+}
+
+// Name returns the label for memory
+func (MemCheck) Name() string {
+	return "Mem"
+}
+
+// Check number of memory required by kubeadm
+func (mc MemCheck) Check() (warnings, errorList []error) {
+	actual := memory.TotalMemory() / 1024 / 1024 // TotalMemory returns bytes; convert to MB
```

Comment:
Since the control plane only runs on Linux it should be fine to use https://golang.org/pkg/syscall/#Sysinfo_t directly without adding another code dependency?

Reply:
ok, let's use https://golang.org/pkg/syscall/#Sysinfo_t
we should add a TODO if *some day* the control-plane is supported on Windows.

Reply:
is there any plan to support control plane on Windows or MacOS?

Reply:
the kubelet does not work on MacOS and there are no plans to support it soon, as far as i know.
control-plane on Windows is not supported and there are no plans to support it anytime soon, but maybe in the future.
the TODO you've added SGTM.